android/app/src/main/java/com/reactnativenavigation/layouts/BaseLayout.java

```diff
@@ -0,0 +1,26 @@
+package com.reactnativenavigation.layouts;
+
+import android.support.v7.app.AppCompatActivity;
+import android.widget.RelativeLayout;
+
+import com.reactnativenavigation.params.SlidingOverlayParams;
+import com.reactnativenavigation.views.slidingOverlay.SlidingOverlay;
+import com.reactnativenavigation.views.slidingOverlay.SlidingOverlaysQueue;
+
+public abstract class BaseLayout extends RelativeLayout implements Layout {
```

Comment:
we should prefer encapsulation over inheritance. If anything, this should be called `SlidingOverlayLayout` but this is yucky :)